documentation/config_logging_advanced.md

```diff
@@ -0,0 +1,130 @@
+
+### Configuring Puppet Server to log to JSON
+
+There are a handful of considerations you may want to take into account before configuring Puppet Server to log to JSON:
+
+* Do you want to configure Puppet Server to *only* log to JSON?  In lieu of the default plain-text logging?  Or do you want to add JSON logging *in addition to* the default plain-text logging?
+* Do you want to set up JSON logging only for the main Puppet Server logs (i.e. `puppetserver.log`), or also for the HTTP access logs (i.e. `puppetserver-access.log`)?
+* What kind of log rotation strategy do you want to use for the new JSON log files?
+
+In this section we'll show some examples of setting up your configuration for logging both the main logs and the HTTP access logs in JSON, *in addition to* leaving the main text-based log files in place.  We'll also configure logback to do log rotation on these new JSON files.  Hopefully you can modify these examples to suit your own needs, based on your answers to the three questions above.
+
+**NOTE**: Puppet Server's packaging sets up `logrotate` management of the main `puppetserver.log` file automatically, which is why you don't see any of the `logback` support for log rotation configured in our default `logback.xml` file.  However, the Puppet Server packaging does *not* include any `logrotate` integration for these new JSON log files that we'll be setting up, so that is why we will show examples of how to use `logback` to do that.  If you were so inclined, you could omit those portions of these configuration examples, and configure `logrotate` to manage the log rotation for your new JSON files.
+
+#### Adding a JSON version of the main Puppet Server logs
+
+To configure Puppet Server to log its main logs to a second log file, in JSON format, simply add a section like this to your `logback.xml` file (it should be added at the same level in the XML as your existing appender/appenders, but the order of the appenders does not matter):
+
+``` xml
+<appender name="JSON" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>/var/log/puppetlabs/puppetserver/puppetserver.log.json</file>
+
+        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+            <fileNamePattern>/var/log/puppetlabs/puppetserver/puppetserver.log.json.%d{yyyy-MM-dd}</fileNamePattern>
+            <maxHistory>5</maxHistory>
+        </rollingPolicy>
+
+        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
+            <fieldNames>
+                <timestamp>timegenerated</timestamp>
+                <message>logmsg</message>
+            </fieldNames>
+        </encoder>
```

Comment:
IMHO this whole block can be skipped. The encoder will generate proper json-formatted events for logstash consumption by default. We added this because of custom processing we do in our logstash setup.